FreeRTOS_IP.c

```diff
@@ -1346,22 +1346,22 @@ void FreeRTOS_SetAddressConfiguration( const uint32_t * pulIPAddress,
         IPStackEvent_t xStackTxEvent = { eStackTxEvent, NULL };
 
         uxTotalLength = uxNumberOfBytesToSend + sizeof( ICMPPacket_t );
-        pxNetworkBuffer = pxGetNetworkBufferWithDescriptor( uxTotalLength, uxBlockTimeTicks );
+        BaseType_t xEnoughSpace;
 
-        if( pxNetworkBuffer != NULL )
+        if( uxNumberOfBytesToSend < ( ipconfigNETWORK_MTU - ( sizeof( IPHeader_t ) + sizeof( ICMPHeader_t ) ) ) )
+        {
+            xEnoughSpace = pdTRUE;
+        }
+        else
         {
-            BaseType_t xEnoughSpace;
+            xEnoughSpace = pdFALSE;
+        }
 
-            if( uxNumberOfBytesToSend < ( ipconfigNETWORK_MTU - ( sizeof( IPHeader_t ) + sizeof( ICMPHeader_t ) ) ) )
-            {
-                xEnoughSpace = pdTRUE;
-            }
-            else
-            {
-                xEnoughSpace = pdFALSE;
-            }
+        if( ( uxGetNumberOfFreeNetworkBuffers() >= 4U ) && ( uxNumberOfBytesToSend >= 1U ) && ( xEnoughSpace != pdFALSE ) )
```

Comment:
I am curious, why was this (3 or 4) count of free network buffers chosen? And not 1 or 2?

Reply:
The number was 3, which was right after allocating a buffer. Now we check the number of buffer before allocating one.
The number was 3 because the IP-task needs at most 3 network buffers to handle a request.

Reply:
Thanks for the explanation @htibosch.